src/main/scala/org/constellation/snapshotstreaming/SnapshotProcessor.scala

```diff
@@ -135,7 +155,8 @@ object SnapshotProcessor {
                 case Some(last) if Validator.isNextSnapshot(last, snapshot.signed.value) =>
                   s3DAO.uploadSnapshot(snapshot) >>
                     prepareAndExecuteBulkUpdate(state, hasher) >>
-                    lastIncrementalGlobalSnapshotStorage.set(snapshot, snapshotInfo)
+                    lastIncrementalGlobalSnapshotStorage.set(snapshot, snapshotInfo) >>
+                    setLastCurrencySnapshots(globalSnapshotWithState.currencySnapshots, hasher)
```

Comment:
could this skip updating the last currency storage if there is an error encountered during the global snapshot step (i.e. is this atomic)? Or is that not a concern in this case?

Reply:
Good catch. I think it is safer to revert the order and set currency snapshots first :+1:

Reply:
I thought about this but actually this way or reverted does not matter. In case of failure, we would end up with partial data indexed anyway. If we check the snapshot chain, it won't allow setting the same snapshot twice, so it won't be possible to reindex missing data automatically. Am I right?

Reply:
Not really @marcinwadon

If saving global succeeds but saving currencies fails then currencies are stuck forever because there will be a "gap" in the chain in currency snapshots after the process restart. Of course, until you manually fix the last currency snapshots or override the global one to the one that failed for currencies.

If we reverse the order then if saving currencies succeeds but global fails then after the restart we do not need to take any special actions. The currencies in the worst case will be indexed twice but at least the chain will be fine.

Reply:
I thought we check the chain validity for currency snapshots too 🤔 If not, then you're right